src/redisearch_rs/trie_rs/src/iter/range.rs

```diff
@@ -0,0 +1,365 @@
+                                continue;
+                            }
+                            Ordering::Equal => {
+                                // The maximum is identical to the current label.
+                                // All the descendants of this node are greater than the maximum.
+                                if !self.is_max_included {
+                                    yield_current = false;
+                                }
+                                visit_descendants = false;
+                            }
+                            Ordering::Greater => {
+                                // The current label is a prefix of the maximum.
+                                // We need to compare the descendants against the remaining prefix.
+                                child_max = Some(&max[node.label_len() as usize..]);
+                            }
+                        },
+                    }
+                }
+
+                if visit_descendants {
+                    self.stack.reserve(node.children().len());
+
+                    let mut max_index = node.children().len();
+                    if let Some(max) = child_max {
+                        if let Some(first) = max.first() {
+                            let i = match node.children_first_bytes().binary_search(first) {
+                                Ok(i) => {
+                                    self.stack.push(StackEntry {
+                                        node: &node.children()[i],
+                                        was_visited: false,
+                                        min: child_min,
+                                        max: child_max,
+                                    });
+                                    i
+                                }
+                                Err(i) => i,
+                            };
+                            max_index = i;
+                        }
+                    }
+
+                    let mut min_index = 0;
+
+                    let mut min_entry = None;
+                    if let Some(min) = child_min {
+                        if let Some(first) = min.first() {
+                            match node.children_first_bytes()[..max_index].binary_search(first) {
+                                Ok(i) => {
+                                    min_entry = Some(StackEntry {
+                                        node: &node.children()[i],
+                                        was_visited: false,
+                                        min: child_min,
+                                        max: None,
+                                    });
+                                    min_index = i + 1;
+                                }
+                                Err(i) => {
+                                    min_index = i;
+                                }
```

Comment:
nitpick - consider aligning the `binary_search` matching of the min and max.
Any reason not to do `xxx_index = match.children_first_bytes()...` in both cases?